daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -827,7 +826,7 @@ private[lf] object Speedy {
         validating: Boolean = false,
         traceLog: TraceLog = newTraceLog,
         warningLog: WarningLog = newWarningLog,
-        contractKeyUniqueness: ContractByKeyUniquenessMode = ContractKeyUniquenessMode.On,
+        contractKeyUniqueness: ContractKeyUniquenessMode = ContractKeyUniquenessMode.Strict,
```

Comment:
Since we're essentially killing `ContractKeyUniquenessMode.On`, consider to completely remove it from `GlobalKey.scala`, along with the then pointless `ContractByKeyUniquenessMode` trait.

Reply:
Let's do that in a following PR.